pkg/dash/dash.go

```diff
@@ -458,3 +503,48 @@ func enableOpenCensus() error {
 
 	return nil
 }
+
+// findKubeConfig looks for kube config from .kube or provided by user
+func findKubeConfig(logger log.Logger, temporaryKubeConfig kubeconfig.TemporaryKubeConfig) <-chan string {
+	r := make(chan string)
+
+	go func() {
+		defer close(r)
+
+		kubeconfig := clientcmd.NewDefaultClientConfigLoadingRules().GetDefaultFilename()
+
+		if _, err := os.Stat(kubeconfig); err == nil {
+			logger.Infof("using kube config: %v", kubeconfig)
+			r <- kubeconfig
+			return
+		}
+
+		select {
+		case path, ok := <-temporaryKubeConfig.Path:
+			if ok {
+				r <- path
+			}
+		}
```

Comment:
No need for the select here.

    path, ok := <-temporaryKubeConfig.Path
    if ok {
        r <- path
    }